Add HWE defconfig

rucoder opened the pull request:
Now kernel config have a flavor. We need it to build the same kernel version with different configs for Evaluation EVE with bigger kernel

- rename eve_defconfig to eve-core_defconfig but keep eve_defconfig symlink for backward compatibility with outdated EVE branches
- introduce new eve-hwe_defconfig with much biggeer set of drivers
- fix incorrect check for CONFIG_BPF_KPROBE_OVERRIDE. it used a local .config copied from host

rene commented on .dockerignore:
```diff
@@ -1,4 +1,24 @@
-.git
+.git/
 Makefile.eve
 Dockerfile.gcc
```
I really like changes here, but they are not applied only for this PR, please, put them into a dedicated commit and provide them for other branches as well, including arm64 ones....

rucoder replied:
@rene Done! I also moved a fix for `CONFIG_BPF_KPROBE_OVERRIDE` check into separate commit. Now both can be backported to ARM branches

rene commented on Makefile.eve:
```diff
@@ -1,6 +1,10 @@
 # Title: Dockerfile for building the EVE kernel
-VERSION=$(shell git rev-parse --short=12 HEAD)
+
```
Although evaluation eve is available on for x86 for now, wouldn't be good to add this support to arm64 as well? I don't like the idea that you are introducing a new variable that is not supported for arm64 branches....

rucoder replied:
@rene we can easily backport it to ARM later, it is just a matter of docker tag names and only x86 will use 2 variants for now. let's test it on x86 before we backport it to other branches. doing it for ARM64 now would require too much time to figure out proper config and eve changes are not in place either. and it was never in the scope for this feature

rucoder replied:
@rene I could actually back-port it to ARM but without config yet. and GHA won't build it. it probably makes sense and simplify logic on EVE side to choose proper kernel commit/flavor
UPDATE: @rene I think we just remove 'hwe' flavor from the matrix in GHA and we should be good

rucoder committed:
Update .dockerignore to minimize docker context

Often kernel source tree has a lot of build artifacts if used for both
container and normal build during the development. To minimize docker context
let's ignore a lot of artifacts from the source tree. it speeds up container build

Signed-off-by: Mikhail Malyshev <mike.malyshev@gmail.com>

rucoder committed:
Add HWE defconfig

Now kernel config have a flavor. We need it to build the same kernel version
with different configs for Evaluation EVE with bigger kernel or to introduce
separate 'rt' config for v6.12.x and get rid of separate -rt branch

- rename eve_defconfig to eve-core_defconfig
- keep eve_defconfig as a symlink to eve-core_defconfig to keep backward
compatibility which old eve branches if we ever update kernel while
necessary EVE changes are not yet merged
- introduce new eve-hwe_defconfig with much bigger set of drivers

Signed-off-by: Mikhail Malyshev <mike.malyshev@gmail.com>

rucoder committed:
Fix incorrect check for CONFIG_BPF_KPROBE_OVERRIDE

This check must be done after .config is generated

Signed-off-by: Mikhail Malyshev <mike.malyshev@gmail.com>

rene reviewed (approved):
LGTM

rucoder committed:
Update Linuxkit version to v1.7.0

We do not pull too many images while building kernel but we still should use
the version of Linuxkit that respects docker login and passes credentials to
buildx builder

Signed-off-by: Mikhail Malyshev <mike.malyshev@gmail.com>
